README.md

```diff
@@ -333,6 +335,143 @@ npx nx g sentry --directory=apps/mobile --dsn=http://your-dsn.ingest.sentry.io/1
+
+### getLinkBuilder
+
+It's a function for building URLs based on a base path and optional query parameters.
+Library - `navigation`.
+
+#### Parameters
+
+- `basePath` — the initial URL. It may contain placeholders for dynamic substitution (e.g., `[id]`).
+
+#### Returns
+
+`() => string` or `(args?: TRootParams) => string` - a function that constructs a URL by replacing placeholders in the `basePath`
+with values from an optional args object and appends query parameters.
+
+#### Example of usage
+
+```ts
+export class ItemSearchParams {
+  @Expose()
+  public categoryId?: number;
+
+  @Expose()
+  public tags?: Array<string>;
+}
+
+const navigationConfig = {
+  routes: {
+    items: getLinkBuilder<ItemSearchParams>('/items'),
+  },
+};
+
+// /items
+const allItemsLink = navigationConfig.routes.items();
+// /items?categoryId=1&tags=fiction&tags=newest
+const filteredItemsLink = navigationConfig.routes.items({
+  categoryId: 1,
+  tags: ['fiction', 'newest'],
+});
+```
+
+### getResourcePaths
+
+It's a function that generates an object of URL paths related to a specific resource:
+list, single view, creation, and editing.
+Library - `navigation`.
+
+#### Parameters
+
+- `basePath` - the root path for the resource.
+- `options` (optional) - an object to enable additional paths.
+  - `withCreation` (optional) includes a path for creation a new resource.
+  - `withEditing` (optional) includes a path for editing an existing resource.
+
+#### Returns
+
+`ResourcePaths<TRootParams>` - an object containing URL path builders for various resource operations:
+
+- `list` - a function to generate the listing URL, supporting query parameters like a result of [getLinkBuilder](#getlinkbuilder).
+- `view` - a function to generate a URL for viewing a specific resource by its id.
+- `create` (optional) - URL string for the creation page, if `withCreation` is enabled.
+- `edit` (optional) - a function to generate a URL for editing a specific resource by its id, if `withEditing` is enabled.
+
+#### Example of usage
+
+```ts
+export class ItemSearchParams {
+  @Expose()
+  public categoryId?: number;
+
+  @Expose()
+  public tags?: Array<string>;
+}
+
+const navigationConfig = {
+  routes: {
+    items: getRecourcePaths<ItemSearchParams>('/items', {
+      withCreation: true,
+      withEditing: true,
+    }),
+  },
+};
+
+const allItemsLink = navigationConfig.routes.items.list(); // /items
+const filteredItemsLink = navigationConfig.routes.items.list({
+  // /items?categoryId=1&tags=fiction&tags=newest
+  categoryId: 1,
+  tags: ['fiction', 'newest'],
+});
+const viewLink = navigationConfig.routes.items.view(1); // /items/1
+const creationLink = navigationConfig.routes.items.create; // /items/create
+const editingLink = navigationConfig.routes.items.edit?.(1); // /items/1/edit
+```
+
+### useFilteringSearchParams (web only)
+
+It's a hook, which converts query parameters to and from a specified model for filtering purposes.
```

Comment:
It worth to mention that this utility uses `useSearchParams` from `next/navigation` under the hood. Because usage of `useSearchParams` in Next.js [requires Suspense ](https://nextjs.org/docs/messages/missing-suspense-with-csr-bailout)